### PR TITLE
ref(ui): Refactor event message/annotations to styled components

### PR DESCRIPTION
**Pull request by billyvg**
This refactors the header message into styled components. This also fixes the hanging annotation.

This affects:

* Issue Details
  * Old: image
  * New:
image
image

* IssueLink hovercard (e.g. on Activity)
image

* CompactIssue (UserFeedback, Activity)
image

* Shared Issue
image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationGroupDetails/header.jsx [hunk 1]
```diff
@@ -103,32 +102,37 @@ class GroupHeader extends React.Component {
...
+                          pathname: baseUrl,
+                          query: {query: 'logger:' + group.logger},
+                        }}
+                      >
+                        {group.logger}
+                      </Link>
+                    </EventAnnotation>
+                  )}
+                  {group.annotations.map((annotation, i) => {
+                    return (
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/header.jsx]: tiny nit, can save one level of indentation here with implicit return.

File: src/sentry/static/sentry/app/components/issueLink.jsx [hunk 2]
```diff
@@ -114,3 +123,40 @@ export default class IssueLink extends React.Component {
...
+    color: ${p => p.theme.gray2};
+    font-size: 90%;
+  }
+`;
+
+const Section = styled('section')`
+  margin-bottom: ${space(2)};
+`;
+
+const Grid = styled('div')`
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/issueLink.jsx]: 🙏 billy on the grid train

File: src/sentry/static/sentry/app/components/events/eventMessage.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        {level}
+      </StyledErrorLevel>
+      {message && <Message>{message}</Message>}
+      {annotations}
+    </div>
+  );
+}
+
+EventMessage.propTypes = {
+  level: PropTypes.oneOf(['error', 'fatal', 'info', 'warning', 'sample']),
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/eventMessage.jsx]: Required?

File: src/sentry/static/sentry/app/components/events/errorLevel.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import styled from 'react-emotion';
+
+const DEFAULT_SIZE = '13px';
+
+function getLevelColor({level, theme}) {
+  const COLORS = {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/errorLevel.jsx]: Can you pull the constant up into the module scope

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/events/errorLevel.jsx (hunk 4 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/events/errorLevel.jsx]: it maps level -> theme

File: src/sentry/static/sentry/app/views/organizationGroupDetails/header.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/header.jsx]: can we add an eslint rule for this

File: src/sentry/static/sentry/less/shared-components.less [hunk 5]
```diff
@@ -1954,30 +1954,6 @@ ul.radio-inputs {
...
-}
-&.level-error .error-level {
-  background: @orange;
-}
-&.level-fatal .error-level {
-  background: @red;
-}
-&.level-sample .error-level {
-  background: @purple;
-}
```
Comment (markstory) [inline src/sentry/static/sentry/less/shared-components.less]: Nice :clap:

File: src/sentry/static/sentry/app/views/organizationGroupDetails/header.jsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/header.jsx]: https://eslint.org/docs/rules/arrow-body-style yup. WIll add

File: src/sentry/static/sentry/app/components/events/eventMessage.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/events/eventMessage.jsx]: It's optional (updated component to reflect it)

**Pull request merged**